driver/src/main/java/org/neo4j/driver/internal/InternalRecord.java

```diff
@@ -41,11 +42,19 @@
     private final Value[] values;
     private int hashcode = 0;
 
-    public InternalRecord( List<String> keys, Map<String, Integer> keyIndexLookup, Value[] values )
+    public InternalRecord( List<String> keys, Value[] values )
     {
         this.keys = keys;
-        this.keyIndexLookup = keyIndexLookup;
         this.values = values;
+
+        int numFields = keys.size();
+        Map<String, Integer> fieldLookup = new HashMap<>( numFields );
+        for ( int i = 0; i < numFields; i++ )
+        {
+            String name = keys.get(i);
+            fieldLookup.put( name, i );
+        }
+        this.keyIndexLookup = fieldLookup;
```

Comment:
for the field-counts we're dealing with (10s), keys.indexOf() should be good enough